modules/consensus/block_validation.go

```diff
@@ -63,7 +64,7 @@ func checkTarget(b types.Block, target types.Target) bool {
 // ValidateBlock validates a block against a minimum timestamp, a block target,
 // and a block height. Returns nil if the block is valid and an appropriate
 // error otherwise.
-func (bv stdBlockValidator) ValidateBlock(b types.Block, minTimestamp types.Timestamp, target types.Target, height types.BlockHeight) error {
+func (bv stdBlockValidator) ValidateBlock(b types.Block, minTimestamp types.Timestamp, target types.Target, height types.BlockHeight, log *persist.Logger) error {
```

Comment:
The reasoning for logging inside `ValidateBlock` is that we're already computing the block size here, doing it in the caller would incur another Marshal.